Xplat/src/main/java/vazkii/botania/test/item/lens/EntropicWarpLensTest.java

```diff
@@ -0,0 +1,40 @@
+package vazkii.botania.test.item.lens;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.gametest.framework.GameTest;
+import net.minecraft.gametest.framework.GameTestHelper;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.block.RedstoneLampBlock;
+import vazkii.botania.common.block.ForceRelayBlock;
+import vazkii.botania.common.block.block_entity.BotaniaBlockEntities;
+import vazkii.botania.common.item.BotaniaItems;
+import vazkii.botania.test.TestingUtil;
+
+public class EntropicWarpLensTest {
+
+	@GameTest(template = "botania:item/lens/entropic_warp_lens")
+	public void testWarp(GameTestHelper helper) {
+		var spreaderPos = new BlockPos(3, 3, 3);
+		var lampPos = new BlockPos(2, 4, 3);
+		var buttonPos = new BlockPos(4, 3, 3);
+		var relayPos = new BlockPos(3, 2, 3);
+		var bindPos = new BlockPos(3, 10, 3);
+		var player = helper.makeMockPlayer();
+
+		var data = ForceRelayBlock.WorldData.get(helper.getLevel());
+		data.mapping.put(helper.absolutePos(relayPos), helper.absolutePos(bindPos));
```

Comment:
you could probably do the binding ahead of time, instead of doing it at test run time. unless the binding doesn't work because of position stuff?

Reply:
You would just do it after building the structure, before saving it. But i don't know if the coordinates behave or not. You can try it if you want, but if another test does it like this then it's probably not a big deal

Reply:
Oh I tried that before and it didn't work, that's why I did it this way

Reply:
> unless the binding doesn't work because of position stuff?

binding data is stored in absolute coords, tests can be loaded anywhere in the world. So doing the binding at runtime is necessary.